### PR TITLE
fix: calculate transaction fee

### PR DESCRIPTION
**Pull request by Torres-ssf**
Pretty much what the title says. 

The `calculateTransactionFee` method was updated based on the version used by `fuels-wallet`

### CODE REVIEW

**Review by arboleya — COMMENTED** [review]
Firstly, I believe it came from @camsjams Vector PR, but just in case, I will mention it here — there are a couple of lint annotations:
 - https://github.com/FuelLabs/fuels-ts/actions/runs/5575462945

> Lint errors are like: _`xyz is defined but never used`_

Finally, following our last `fuel-gauge` discussion about where to put tests and whatnot, should we move the `fee.test.ts` over there (to `fuel-gauge` since it's not a unit test?

**Comment by Torres-ssf** [comment]
> Firstly, I believe it came from @camsjams Vector PR, but just in case, I will mention it here — there are a couple of lint annotations:
> 
> * https://github.com/FuelLabs/fuels-ts/actions/runs/5575462945
> 
> > Lint errors are like: _`xyz is defined but never used`_
> 
> Finally, following our last `fuel-gauge` discussion about where to put tests and whatnot, should we move the `fee.test.ts` over there (to `fuel-gauge`) since it's not a unit test?

@arboleya but the test inside `fee.test.ts` are unit tests :thinking: 

@nedsalk just suggested to me making them feature test, so instead of having these unit tests on `fee.test.ts`, we would have the value returned by these functions validated in the test that tests the function that makes the call to `calculateTransactionFee`

**Comment by arboleya** [comment]
@Torres-ssf Are they unit tests? Maybe I thought you were querying the node. Perhaps I got something wrong.

**Review by danielbate — APPROVED** [review]
@arboleya LGTM as the tests in that file are the calculations that are passed values that would be from the node. Begs the question as to whether we'd want an E2E that tests for a specific `gasUsed` and `fee` value for a given contract. At a glance most of our tests check that `gasUsed` or `fee` are at least defined. I don't think such a test is necessary for this piece of work so will leave up to @Torres-ssf 👍🏻

**Comment by arboleya** [comment]
@Torres-ssf @danielbate My bad, I mixed things up.

The `fee.test.ts` is indeed a unit.

Unlike the `provider.test.ts` — which uses the node for fetching stuff.

As I said before, I prefer this approach of having the `/test` directory, but that wasn't what we discussed at our last discussion with @camsjams. From what I recall, this test should be on the `fuel-gauge` and import the Provider from `fuels`, like so:

```ts
import { Provider } from 'fuels';
```

Was it me who understood things wrong? I'm not claiming I'm right.

But we surely need to clearly define these convention rules to avoid misinterpretations.

**Comment by danielbate** [comment]
My understanding is that unit tests can remain and will be alongside `src` files. Feature tests, which will be the entry points for the packages, will go in a `/test` directory. Anything making calls to a node and thus requiring encoding/decoding etc will go in `fuel-gauge` as an integration test. On which should be prioritised and implemented is still under discussion / up to the implementer.

I plan on defining this more clearly in #1043 once #1079 has been completed and merged.

Apologies also I missed the `provider.test.ts`, agreed that under definition this would come under an integration test and should be within `fuel-gauge`.

**Comment by Torres-ssf** [comment]
@arboleya @arboleya I've only touched on one test case from `provider.test.ts` to make it conform with the changes that I made.

No test cases were implemented on it.

Do you think that moving `provider.test.ts` to `fuel-gauge` should be part of this PR scope?

**Comment by danielbate** [comment]
@Torres-ssf I'd be happy to count it as out of scope of this work and instead include it in a provider test refactor piece

**Review by arboleya — DISMISSED** [review]

**Comment by arboleya** [comment]
:shipit:

**Comment by Torres-ssf** [comment]
@arboleya @danielbate I've just revet the renaming of a method and query to avoid conflicts with the fuels wallet

**Review by arboleya — APPROVED** [review]

**Review by nedsalk — APPROVED** [review]

**Pull request merged**
